kubernetes/windows/main.ps1

```diff
@@ -288,6 +309,12 @@ Start-Transcript -Path main.txt
 Remove-WindowsServiceIfItExists "fluentdwinaks"
 Set-EnvironmentVariables
 Start-FileSystemWatcher
+
+#Bootstrapping CA certs for non public clouds and AKS clusters
+if (![string]::IsNullOrEmpty([System.Environment]::GetEnvironmentVariable("AKS_RESOURCE_ID"))) {
```

Comment:
Rashmi, Since AKS_RESOURCE_ID env used for both AKS, Azure Arc K8s and ARO cluster. We need to add check for AKS cluster type by checking the cluster resource id has "Microsoft.ContainerService/managedClusters" in it. Please do case insensitive check since the cluster resource id can come different casing.

Reply:
We discussed this in the afternoon and decided to go this way because we want all managed clusters to be supported in these clouds. That said, you need to provide  better tooling/API support for different cluster flavors (ARO,ARC,AKS-E etc..). Just doing this env var check in many many places is not the correct & efficient thing to do. What we need is a fully typed cluster metadata class with properties that identifies cluster attributes that we are interested in.

Reply:
Create task in the oct agent release https://msazure.visualstudio.com/InfrastructureInsights/_sprints/taskboard/Containers/InfrastructureInsights/Iron/20Oct02?workitem=8463970 to handle cluster type detection in unified fashion across all the places. We can discuss about what would be best and consistent, and efficient way to do this.